optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc

```diff
@@ -176,6 +176,31 @@ The following constraint stream cardinalities are currently supported:
 |4          |  Quad|`QuadConstraintStream<A, B, C, D>`
 |===
 
+[[constraintStreamsHigherCardinalities]]
+=== Achieving higher cardinalities
+
+OptaPlanner currently does not support constraint stream cardinalities higher than 4.
+However, with <<constraintStreamsMappingTuples,tuple mapping>> it is easy to achieve effectively infinite cardinality:
+
+[source,java,options="nowrap"]
+----
+    private Constraint pentaStreamExample(ConstraintFactory factory) {
+        return factory.from(Shift.class) // UniConstraintStream<Shift>
+                .join(Shift.class)       // BiConstraintStream<Shift, Shift>
+                .join(Shift.class)       // TriConstraintStream<Shift, Shift, Shift>
+                .join(Shift.class)       // QuadConstraintStream<Shift, Shift, Shift, Shift>
+                .map(Tuple::of)          // UniConstraintStream<Tuple<Shift, Shift, Shift, Shift>>
+                .join(Shift.class)       // BiConstraintStream<Tuple<Shift, Shift, Shift, Shift>, Shift>
```

Comment:
the class Tuple doesn't exist, that makes it a bit clumsy to copy-paste. commons-lang3 has Triple and Pair, but not for 4...